modules/api/src/main/java/org/apache/ignite/compute/IgniteCompute.java

```diff
@@ -244,6 +244,98 @@ default <T, R> Map<ClusterNode, R> executeBroadcast(
         return map;
     }
 
+    /**
+     * Submits a {@link ComputeJob} of the given class for an execution on nodes where the primary replicas of a partitions of the specified
+     * table are located. The partition indices are passed to the job in the {@link JobExecutionContext#partitions()}.
```

Comment:
Is there any "partition pinning"? What if the partition is moved to another node while the job is executing?

Reply:
Then, in my opinion, we don't need those new broadcast methods. Instead let's simplify "get local partitions for a table" scenario by extending `PartitionManager` API.

Thoughts?